server/controllers/board.js

```diff
@@ -0,0 +1,76 @@
+const asyncHandler = require("express-async-handler");
+const { User } = require("../models/User");
+const Board = require("../models/Board");
+
+exports.getBoards = asyncHandler(async (req, res) => {
+  const boards = await Board.find();
+
+  return res.json(boards);
+});
+
+exports.getBoard = asyncHandler(async (req, res) => {
+  try {
+    const board = await Board.findById(req.params.id);
+
+    if (!board) res.status(404).json({ error: "Board not found" });
+
+    return res.status(200).json(board);
+  } catch (error) {
+    return res.json(error);
+  }
+});
+
+// createBoard should be in boards controller + routes instead of users
+exports.createBoard = asyncHandler(async (req, res) => {
+  // This needs to be fixed according to a non-double-stringified body from frontend
+  const { board } = req.body;
+  const temp = JSON.parse(board);
+  const boardValues = JSON.parse(temp);
+
+  try {
+    // @ts-ignore
+    const user = await User.findById(req.user.id);
+
+    if (!user) res.status(404).json({ error: "User not found" });
+
+    const newBoard = await Board.create({
+      name: boardValues.name,
+      columns: boardValues.columns,
```

Comment:
Perhaps we should set "In progress" and "Done" columns by default as the same with the initial board when a user signs up. I suppose we could pass this down in the request by creating the two columns but just a thought. 🤔

Reply:
I'm thinking of creating 3 columns of `Not Started`, `In Progress`, and `Completed`/`Done`. `Not Started` will also begin with a couple of default cards that can display to users what you can do with it like a demo card!